vivisect/parsers/elf.py

```diff
@@ -613,36 +621,56 @@ def loadElfIntoWorkspace(vw, elf, filename=None, baseaddr=None):
             sva += baseoff
             if symname:
                 vw.makeName(sva, symname, filelocal=True, makeuniq=True)
-                valu = vw.readMemoryPtr(sva)
-                if not vw.isValidPointer(valu) and s.st_size == vw.psize:
-                    vw.makePointer(sva, follow=False)
+                if sva == 0x0:
+                    # if the address of the object is 0x0,
+                    # then it is probably an extern object, like the environ pointer provided by glibc.
+                    pass
+                elif not vw.isValidPointer(sva):
+                    # if the address of the object is invalid, 
+                    # this is weird, but we'll try to carry on.
+                    pass
                 else:
-                    '''
-                    Most of this is replicated in makePointer with follow=True. We specifically don't use that,
-                    since that kicks off a bunch of other analysis that isn't safe to run yet (it blows up in
-                    fun ways), but we still want these locations made first, so that other analysis modules know
-                    to not monkey with these and so I can set sizes and what not.
-                    And while ugly, this does cover a couple nice use cases like pointer tables/arrays of pointers being present.
-                    '''
-                    if not valu:
-                        # do a double check to make sure we can even make a pointer this large
-                        # because some relocations like __FRAME_END__ might end up short
-                        psize = vw.getPointerSize()
-                        byts = vw.readMemory(sva, psize)
-                        if len(byts) == psize:
-                            new_pointers.append((sva, valu, symname))
-                    elif vw.isProbablyUnicode(sva):
-                        vw.makeUnicode(sva, size=s.st_size)
-                    elif vw.isProbablyString(sva):
-                        vw.makeString(sva, size=s.st_size)
-                    elif s.st_size % vw.getPointerSize() == 0 and s.st_size >= vw.getPointerSize():
-                        # so it could be something silly like an array
-                        for addr in range(sva, sva+s.st_size, vw.psize):
-                            valu = vw.readMemoryPtr(addr)
-                            if vw.isValidPointer(valu):
-                                new_pointers.append((addr, valu, symname))
+                    try:
+                        valu = vw.readMemoryPtr(sva)
+                    except e_exc.SegmentationViolation:
+                        valu = 0
+
+                    if not vw.isValidPointer(valu) and s.st_size == vw.psize:
+                        vw.makePointer(sva, follow=False)
```

Comment:
i would appreciate a comment here on the logic.  i get that we're making a pointer to a location that doesn't exist in the workspace, which is fine because `follow=False` but what is the logic and some of the use-cases/context behind this?

Reply:
@williballenthin if you can answer this/add a comment, i'm ready to approve this PR

Reply:
this `if` branch was existing code. i just moved it one level deeper of nesting.

Reply:
old line 617

Reply:
As far as I can tell this region of code tries to guess the type of data pointed to by the object symbol, which is reasonable but not foolproof.

Reply:
sorry, i meant "add a comment to the code because it's not obvious and future us-s will thank you.

Reply:
Ok, I understand that you're asking me to add comments to the old/existing code. I will do my best to figure out the original author's intention and summarize it here.

As I'm AFK this week, I expect to have this completed around Aug 5 or 6.

Reply:
@williballenthin bump :)

Reply:
> As I'm AFK this week, I expect to have this completed around Aug 5 or 6.

Reply:
@williballenthin thank you!  will watch for it.  will i see you at hacker summer camp?

Reply:
unfortunately not this year. my partner and I live in Germany with little ones for a few years, so it's tough to make it back. meeting you would be a highlight, though. 😁

Reply:
@atlas0fd00m please review comments and simplifications